packages/nextjs/pages/index.tsx

```diff
@@ -1,24 +1,177 @@
+    contractName:"Balloons",
+    functionName:"balanceOf",
+    args: [accountBalanceOf]
+  });
+
+  const {data: contractBalance} = useScaffoldContractRead({
+    contractName:"Balloons",
+    functionName:"balanceOf",
+    args: [DEXInfo?.address]
+  });
+
+  const { data: contractETHBalance } = useBalance({
+    address: DEXInfo?.address,
+  })
+
   return (
     <>
       <MetaHeader />
-      <div className="flex items-center flex-col flex-grow pt-10">
-        <div className="px-5">
-          <h1 className="text-center mb-8">
+      <h1 className="text-center mb-4 mt-9">
             <span className="block text-2xl mb-2">SpeedRunEthereum</span>
-            <span className="block text-4xl font-bold">Challenge #X: Challenge Title </span>
+            <span className="block text-4xl font-bold">Challenge 4: Minimum Viable Exchange </span>
           </h1>
-          <p className="text-center text-lg">
-            Get started by editing{" "}
-            <code className="italic bg-base-300 text-base font-bold">packages/nextjs/pages/index.tsx</code>
-          </p>
-          <p className="text-center text-lg">
-            Edit your smart contract <code className="italic bg-base-300 text-base font-bold">YourContract.sol</code> in{" "}
-            <code className="italic bg-base-300 text-base font-bold">packages/hardhat/contracts</code>
-          </p>
+      <div className="flex items-center flex-col flex-grow pt-10 grid grid-cols-2">
+        <div className="px-5 border rounded-3xl py-5 ml-3">
+              <div className="flex text-center mb-8 px-4 gap-4">
+                <span className="block text-2xl mb-2"><Address address={DEXInfo?.address}/></span>
+                <span><Balance address={DEXInfo?.address}/></span>
+                {isLoading ? (
+                  <span>Loading...</span>
+                ) : (
+                  <span>Balloons: {ethers.utils.formatEther(DEXBalloonBalance?.toString() || 0)}</span>
+                )}
+              </div>
+              <div className="border rounded-3xl py-3 px-4">
+              <div className="flex mb-4">
+                <span>ethToToken  <EtherInput value={ethToTokenAmount} onChange={(value) => setEthToTokenAmount(value)} name="ethToToken"/></span>
+                <button className="btn btn-primary h-[2.2rem] min-h-[2.2rem] mt-6 mx-5" onClick={ethToTokenWrite}>Send</button>
+              </div>
+              <div className="flex">
+                <span>tokenToETH  <EtherInput value={tokenToETHAmount} onChange={(value) => setTokenToETHAmount(value)} name="tokenToETH"/></span>
+                <button className="btn btn-primary h-[2.2rem] min-h-[2.2rem] mt-6 mx-5" onClick={tokenToEthWrite}>Send</button>
+              </div>
+            </div>
+            <div className="relative flex py-5 items-center">
+                <div className="flex-grow border-t border-gray-400"></div>
+                <span className="flex-shrink mx-4 text-gray-400">Liquidity(None)</span>
```

Comment:
It's not always none, check original https://github.com/scaffold-eth/scaffold-eth-challenges/blob/challenge-4-dex/packages/react-app/src/components/DEX.jsx#L110